storage/csi-snapshot.md

```diff
@@ -0,0 +1,120 @@
+
+## Motivation
+
+* This feature has been requested by both the OCS and CNV teams.
+* Snapshotting is an incredibly useful function when it comes to system changes in development areas, as it allows rapid rollback to preview dev versions
+* Container development benefits from filesystems that provide snapshot functionality
+
+### Goals
+
+* Backport and include the external-snapshotter sidecar.
+
+* Package and ship a downstream image of the this sidecar.
+
+### Non-Goals
+
+* Offer application-consistent snapshots by providing pre/post snapshot hooks to freeze/unfreeze applications and/or unmount/mount file system.
+
+* Provide higher-level management, such as backing up and restoring a pod and statefulSet, and creating a consistent group of snapshots.
+
+## Proposal
+
+This feature has already been implemented upstream, and is described in https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/20190709-csi-snapshot.md . It is currently documented upstream at https://kubernetes.io/docs/concepts/storage/volume-snapshots/ .
+
+We will be releasing the `external-snapshotter` sidecar image to enable CSI Snapshot support in OCP as Developer Preview only. This sidecar can be used by internal teams, such as CNV or OCS, for their drivers.
+
+### User Stories [optional]
+
+Similar to how users and administrators provision PersistentVolume and PersistentVolumeClaim resources, VolumeSnapshotContent and VolumeSnapshot API resources will be included to create volume snapshots for users and administrators. A VolumeSnapshotContent is a snapshot taken from a volume in the cluster that has been provisioned by an administrator. It is a resource in the cluster, similar to a PersistentVolume. A VolumeSnapshot is a request for snapshot of a volume by a user. It is similar to a PersistentVolumeClaim.
+
+In addition to the above, VolumeSnapshotClass resources can be used to describe the “class” of storage when provisioning a VolumeSnapshot.
+
+#### Story 1
+
+As an OCS developer, I want to release the `external-snapshotter` sidecar image so that CSI drivers can utilize volume snapshots.
+
+#### Story 2
+
+### Implementation Details/Notes/Constraints [optional]
+
+This feature has already been implemented upstream. Therefore, this feature requires a Kubernetes rebase of 1.15 or later.
+
+### Risks and Mitigations
+
+* Kubernetes 1.15+ rebase
+
+## Design Details
+
+### Test Plan
+
+We’ll want the e2e tests to run using CSI sidecars shipped as part of OCP. There is ongoing progress in the following links regarding testing OCP CSI sidecars:
+
+* https://github.com/openshift/origin/pull/23560
+* https://jira.coreos.com/browse/STOR-223
+Tests for this sidecar are currently defined upstream at https://github.com/kubernetes-csi/external-snapshotter/tree/master/pkg/controller .
+
+### Graduation Criteria
+
+##### Dev Preview -> Tech Preview
+
+Feature complete, including:
+Create/delete volume snapshots
```

Comment:
This is not rendered well by github. Perhaps it should be reworked to bulleted list.

Reply:
Corrected the syntax to include bullets.